_posts/2020-05-26-exploring-the-open-sgid-a-brief-tutorial-using-open-source-python-tools.md

```diff
@@ -0,0 +1,169 @@
+sql_faults = "select * from opensgid.geoscience.quaternary_faults"
+faults = gpd.GeoDataFrame.from_postgis(sql_faults, con, geom_col='shape')
+
+# Filter down to faults longer than 1 mile, create 'length_mi' column
+long_faults = faults[faults['shape'].length > 1609.34].copy()
+long_faults['length_mi'] = long_faults['shape'].length/1609.34
+
+# Plot faults symbolized by length of the fault
+fig, ax = plt.subplots(figsize=(6, 10))
+counties.plot(ax=ax, color='gray', edgecolor='black')
+long_faults.plot(ax=ax, column='length_mi', cmap='viridis', vmin=0, vmax=30, legend=True,
+                 legend_kwds={'label': 'Fault Length [miles]', 'shrink': 0.75})
+plt.title('Utah Faults > 1 Mile')
+plt.show()
+```
+
+[![Utah Faults > 1 Mile]({% link images/EMN_20200526_faults_plot.png %} "Utah Faults > 1 Mile")]({% link images/EMN_20200526_faults_plot.png %}){: .flex .flex--center }
+
+## EXAMPLE 4: Libraries in State Senate District 24
+
+For this fourth example, you'll plot public libraries within State Senate District 24 by using a spatial operation, which is really where GeoPandas shines and sets itself apart from nonspatial tools.
+1. Start by selecting the data layers (hopefully the syntax is getting familiar now) with SQL queries and importing them with GeoPandas.
+1. Next, subset the senate districts down to just district 24 using Pandas square bracket `[]` notation.
+1. Then, subset the libraries data set by selecting the libraries (and their geometry) that are within district 24.
+1. Finally, plot all districts, the selected district (24), and all libraries that fall within the selected district.
+
+```py
+# Utah Libraries in Senate District 24
+sql_sen = "select * from opensgid.political.senate_districts_2012"
+sen_districts = gpd.GeoDataFrame.from_postgis(sql_sen, con, geom_col='shape')
+
+sql_lib = "select * from opensgid.society.libraries"
+libraries = gpd.GeoDataFrame.from_postgis(sql_lib, con, geom_col='shape')
+
+# Filter down to district 24 and select libraries within the district
+dist_24 = sen_districts.loc[sen_districts['dist'] == 24]
+lib_24 = libraries[libraries.within(dist_24.unary_union)]
+
+# Plot senate districts and libraries
+fig, ax = plt.subplots(figsize=(6, 10))
+sen_districts.plot(ax=ax, color='lightgray', edgecolor='black')
+dist_24.plot(ax=ax, color='skyblue', edgecolor='black')
+lib_24.plot(ax=ax, color='red', markersize=10)
+plt.title('Libraries in Utah Senate District 24')
+plt.show()
+```
+
+[![Libraries in Utah Senate District 24]({% link images/EMN_20200526_libraries_plot.png %} "Libraries in Utah Senate District 24")]({% link images/EMN_20200526_libraries_plot.png %}){: .flex .flex--center }
+
+That's it! This brief tutorial has shown a few different ways to gather and analyze data from the Open SGID with the psycopg2 and GeoPandas libraries. Hopefully it was helpful and will spark your interest in both the Open SGID and using open source Python tools.
+
+* [AGRC]: Utah Automated Geographic Reference Center
```

Comment:
**suggestion**: you might need another blank line for the abbreviations to kick in

Reply:
done